lib/cadet/assessments/mission.ex

```diff
@@ -8,19 +8,32 @@ defmodule Cadet.Assessments.Mission do
 
   alias Cadet.Assessments.Category
   alias Cadet.Assessments.Image
+  alias Cadet.Assessments.Question
 
   schema "missions" do
-    field(:order, :string)
-    field(:category, Category)
     field(:title, :string)
+    field(:is_published, :boolean, default: false)
+    field(:category, Category)
     field(:summary_short, :string)
     field(:summary_long, :string)
     field(:open_at, Timex.Ecto.DateTime)
     field(:close_at, Timex.Ecto.DateTime)
+    field(:max_xp, :integer, default: 0)
+    field(:priority, :integer, default: 0)
     field(:cover_picture, Image.Type)
+    has_many(:questions, Question, on_delete: :delete_all)
+    field(:order, :string, default: "")
+    timestamps()
   end
 
-  @required_fields ~w(order category title open_at close_at)a
-  @optional_fields ~w(summary_short summary_long)
+  @required_fields ~w(category title open_at close_at)a
+  @optional_fields ~w(summary_short summary_long is_published max_xp priority)a
   @optional_file_fields ~w(cover_url)
+
+  def changeset(mission, params) do
+    mission
+    |> cast(params, @required_fields ++ @optional_fields)
+    |> validate_required(@required_fields)
+    |> validate_number(:max_xp, greater_than_or_equal_to: 0)
```

Comment:
You need to process the `cover_url`

Reply:
I am not exactly sure how to process this.